src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -0,0 +1,316 @@
+        {
+            this.writer.WriteEndObject();
+        }
+
+        public void StartArrayScope()
+        {
+            this.writer.WriteStartArray();
+        }
+
+        public void EndArrayScope()
+        {
+            this.writer.WriteEndArray();
+        }
+
+        public void WriteName(string name)
+        {
+            this.writer.WritePropertyName(name);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(bool value)
+        {
+            this.writer.WriteBooleanValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(int value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(float value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(short value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(long value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+            
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(double value)
+        {
+            this.writer.WriteNumberValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Guid value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(decimal value)
+        {
+            if (this.isIeee754Compatible)
+            {
+                this.writer.WriteStringValue(value.ToString(CultureInfo.InvariantCulture));
+            }
+            else
+            {
+                this.writer.WriteNumberValue(value);
+            }
+
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(DateTimeOffset value)
+        {
+            this.writer.WriteStringValue(value);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeSpan value)
+        {
+            string stringValue = EdmValueWriter.DurationAsXml(value);
+            this.writer.WriteStringValue(stringValue);
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(Date value)
+        {
+            this.writer.WriteStringValue(value.ToString());
+            this.FlushIfBufferThresholdReached();
+        }
+
+        public void WriteValue(TimeOfDay value)
+        {
+            this.writer.WriteStringValue(value.ToString());

```

Comment:
Do we need to call `this.FlushIfBufferThresholdReached()` after writing the `TimeOfDay` value?